src/libsync/syncfilestatustracker.cpp

```diff
@@ -187,12 +187,13 @@ void SyncFileStatusTracker::incSyncCountAndEmitStatusChanged(const QString &rela
 
         // We passed from OK to SYNC, increment the parent to keep it marked as
         // SYNC while we propagate ourselves and our own children.
-        ASSERT(!relativePath.endsWith('/'));
```

Comment:
this one helped me detect a real bug
we should not remove this check
it is about making sure we do not try to propagate twice the same folder:
`folder` and `folder/` are different but point to the same real folder
having both of them in the propagation queue is an error that should be blocked/prevented

Reply:
Sure, I'll replace the assert with one that checks that the two paths are the same but one has trailing slash